openpype/modules/deadline/plugins/publish/submit_publish_job.py

```diff
@@ -733,6 +736,13 @@ def _get_representations(self, instance, exp_files, do_not_add_review):
             if not already_there:
                 representations.append(rep)
 
+        for rep in representations:
+            # inject colorspace data
+            self.set_representation_colorspace(
+                rep, self.context,
+                colorspace=instance_data["colorspace"]
```

Comment:
Wouldn't it be safer to do `instance_data.get` and default to something?

Reply:
Agreed - please lets make this optional.

Reply:
It is filled on ln 875, so the key will always be there. 
`set_representation_colorspace` handles case where colorspace is None, in that case it uses File rules and `get_imageio_colorspace_from_filepath` to get some value, so I don't think it is necessary to provide any default value here.